Sprint scm integration (#302)
openSUSE/obs-landing: _posts/development/2021-07-12-scm-integration-more-improvements.md

---
layout: post

> title: More Improvements for the Continuous Integration with OBS and GitHub/GitLab

Reviewer: The next one will be `more more improvements`? ;-)

> title: More Improvements for the Continuous Integration with OBS and GitHub/GitLab

Reviewer: ```suggestion
title: Reporting filters and on-premise support for continuous integration
```

Author: `on-premise` is not the wrong word here...

category: development
---

> Today, based on some new feedback from those of you that enrolled in our [beta program](/2018/10/04/the-beta-program/), we have more improvements for the continuous integration we unveiled in our [previous blog post](/2021/05/31/scm-integration/).

Reviewer: Please make sure you also include links to all the other posts in this series. In the other PR they created a partial

https://github.com/openSUSE/obs-landing/pull/304/files#diff-b91ac19286f2f49c0aacacb76f4be0abe661c6a209b7530e67133d9ac551055fR10

> ## Limit what we report to the SCM:

Reviewer: ```suggestion
## Limit what is reported to the pull/merge request
```

> Some OBS projects have dozens of repositories and architecture enabled, but maybe the users just want to have some repositories being reported back to the SCM as status check. Therefore users are going to be able to limit repositories and architectures using [Filters](https://github.com/openSUSE/open-build-service/wiki/Better-SCM-CI-Integration#filters-in-workflows)

Reviewer: ```suggestion
Some of your OBS projects probably have dozens of repositories and architectures enabled. But maybe you just want to have some of them reported back to the pull/merge request? Therefore you are now able to limit repositories and architectures using the new filters features.
```

Reviewer: And please do not link to the developer wiki. This blog post is the place to tell people how to setup filters.

Author: > And please do not link to the developer wiki. This blog post is the place to tell people how to setup filters.

Because the whole thing still under development, this blog post will become quite soon out of date, beside that we link quite often our wiki in our posts

Reviewer: If this blog post gets out of date then we can update it. We have linked a couple of times to the developer wiki when the post was about OBS developer stuff. This is a feature for our customers, not for OBS developers.

> ## Support private instances from gitlab and github:

Author: > I think the term is `on-premise`

Not really, On-premises is the software or technology that is located within the physical confines of an enterprise often in the company’s data center as opposed to running remotely on hosted servers or in the cloud.

That's not the case here.

> ## Support private instances from gitlab and github:

Author: From gitlab documentation `privately-hosted GitLab instance`

> ## Support private instances from gitlab and github:

Author: other term that comes often - including in the "installer" - is `Self-Hosted`

> ## Support private instances from gitlab and github:

Reviewer: I'm fine with anything that makes the distinction more clear.

> ## Support private instances from gitlab and github:

Reviewer: self-hosted it is then! I like it.

After collecting feedback, we prioritized the support to private instances from Gitlab and Github. From now one, you are able to configure your private SCM with OBS (as long as the connectivity works, OBS will be able to interact with that SCM). 

# Current limitation

> Today, users have to make sure that their SCM have SSL well configured using a trusted certificate. The user cannot neither install a custom SSL certificate or ignore it

Author: How a b.o.o user is able to install a root certificate for her self-signed digital certificates to be able to enable OBS to download the `workflows.yml` from her gitlab.example.org? It is not possible, unless the certificate is installed in the operating system layer... 

related: https://github.com/openSUSE/open-build-service/issues/11316

> Today, users have to make sure that their SCM have SSL well configured using a trusted certificate. The user cannot neither install a custom SSL certificate or ignore it

Reviewer: A b.o.o user can not make the SSL certificate trusted for sure. If you have your own SCM and your own OBS you can.

> Today, users have to make sure that their SCM have SSL well configured using a trusted certificate. The user cannot neither install a custom SSL certificate or ignore it

Reviewer: I still think this warrants an update to the error post about connection problems. If the OBS and the SCM installation can not talk to each other all of this won't work. Reasons they can't talk to each other:

- SSL certificate verification problems
- Network problems
- Private repositories on gitlab

and more possibly...

---
layout: post
title: Reporting Filters and Self-Hosted support for Continuous Integration 
category: development
---

Today, based on some new feedback regarding the SCM integration, we have some news for those of you that enrolled in our [beta program](/2018/10/04/the-beta-program/) 

{% include partials/_series-of-posts-about-scm-integration.md %}

## Limit what is reported to the pull/merge request

Some of your OBS projects probably have dozens of repositories and architectures enabled. But maybe you just want to have some of them being reported back to the SCM as status check to the pull/merge request. Therefore you are now able to limit repositories and architectures using the new filters features.

### Filters

> Filters are the way to limit which build results are being reported back to the SCM. Keep in mind that Filters are going to evolute to support many other cases, but for now, we reduced the scope to architectures and repositories only.

Reviewer: ```suggestion
Filters are the way to limit which build results are being reported back to the SCM. Keep in mind that Filters are going to evolve to support many other cases, but for now, we reduced the scope to architectures and repositories only.
```

In the `workflows.yml` you can use the following syntax to accomplish that:

```
workflow:
  steps:
    - branch_package:
        source_project: OBS:Server:Unstable
        source_package: obs-server
  filters:
    architectures:
      only:
        - x86_64
    repositories:
      only:
        - openSUSE_Factory
        - openSUSE_15.1
        - SLE_15_SP1
        - SLE_15_SP3
```

> `only` and `ignore` are your tools to control what and what not are going to be reported back as status check.

Reviewer: ```suggestion
`only` and `ignore` are your tools to control what and what not are going to be reported back as status check. Only has precedence over ignore, so if both are defined, ignore is not considered.
```
